### PR TITLE
chore: Add method to verify the prestate in the execution witness

### PR DESCRIPTION
**Pull request by kevaundray**
This adds a method that verifies the proof in the execution witness.

It takes in:

- The prestate root as a string in hex, with 0x prefixed 
- The execution witness as a string, exactly as it is in the block.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "@typescript-eslint/eslint-plugin": "^6.7.3",
     "@typescript-eslint/parser": "^6.7.3",
     "@vitest/browser": "^1.4.0",
+    "babel-cli": "^6.26.0",
```
Comment (kevaundray) [inline package.json]: Most of the diff is from this. My local laptop was not able to build locally without npm re-installing this. Will remove soon, as I'm assuming it works in CI

File: src.ts/index.ts [hunk 2]
```diff
@@ -26,13 +27,17 @@ export const loadVerkleCrypto = async (): Promise<VerkleCrypto> => {
   ): Commitment =>
     updateCommitmentBase(verkleFFI, commitment, commitmentIndex, oldScalarValue, newScalarValue)
 
+  const verifyExecutionWitnessPreState = (prestateRoot: string, execution_witness_json: string): boolean =>
```
Comment (kevaundray) [inline src.ts/index.ts:30]: This is essentially doing nothing, wasn't sure if you wanted to keep this so it follows the same pattern as everything else. Will remove if so

File: src.rs/Cargo.toml [hunk 3]
```diff
@@ -15,11 +15,11 @@ default = ["console_error_panic_hook"]
 serde = { version = "1.0", features = ["derive"] }
 serde-wasm-bindgen = "0.3.0"
 wasm-bindgen = { version = "0.2.90", features = ["serde-serialize"] }
-verkle-trie = { git = "https://github.com/crate-crypto/rust-verkle", rev = "7688f0aedfb147d3d391abfe8495e46c46d72ce0" }
-verkle-spec = { git = "https://github.com/crate-crypto/rust-verkle", rev = "7688f0aedfb147d3d391abfe8495e46c46d72ce0" }
-ipa-multipoint = { git = "https://github.com/crate-crypto/rust-verkle", rev = "7688f0aedfb147d3d391abfe8495e46c46d72ce0" }
-banderwagon = { git = "https://github.com/crate-crypto/rust-verkle", rev = "7688f0aedfb147d3d391abfe8495e46c46d72ce0" }
-ffi_interface = { git = "https://github.com/crate-crypto/rust-verkle", rev = "7688f0aedfb147d3d391abfe8495e46c46d72ce0" }
+verkle-trie = { git = "https://github.com/crate-crypto/rust-verkle", rev = "594a70e3df16db9d870b7d3c82479d041cfd1b2d" }
```
Comment (kevaundray) [inline src.rs/Cargo.toml]: This has been pulled out into #45

File: src.ts/index.ts (hunk 2 above)
Comment (acolytec3) [reply src.ts/index.ts:30]: What's the difference between this function and what is being done in [this PR](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3419)?

Comment (gabrocheleau) [reply src.ts/index.ts:30]: The difference is that my ethereumjs PR is verifying that the prestate is minimal (i.e. it contains no extra data in the witness), whereas this verifies and proves that the prestate belongs the the parent state root.

**Review by gabrocheleau — APPROVED** [review]
Thank you very much for this @kevaundray !

I've modified the test data so that it taps into existing kaustinen6 blocks. The proof verification works with one of the test blocks, but not with one of the larger ones (~100 txs and a massive witness). I've commented out that second one for now.

**Comment by holgerd77** [comment]
> but not with one of the larger ones (~100 txs and a massive witness)

@gabrocheleau what happens on the larger ones? Timing out, or a concrete error message (in that case, can you post)?

**Comment by gabrocheleau** [comment]
> > but not with one of the larger ones (~100 txs and a massive witness)
> 
> @gabrocheleau what happens on the larger ones? Timing out, or a concrete error message (in that case, can you post)?

here is the log:

```
 FAIL  src.ts/tests/ffi.spec.ts > bindings > verifyExecutionProof: block with many txs
RuntimeError: unreachable
 ❯ __rust_start_panic wasm:/wasm/00264a8e:1:483948
 ❯ rust_panic wasm:/wasm/00264a8e:1:483012
 ❯ std::panicking::rust_panic_with_hook::hbf46ef0245cc9589 wasm:/wasm/00264a8e:1:453971
 ❯ std::panicking::begin_panic_handler::{{closure}}::hc07db454214d2c87 wasm:/wasm/00264a8e:1:465026
 ❯ std::sys_common::backtrace::__rust_end_short_backtrace::hf9e2f055fb5ef672 wasm:/wasm/00264a8e:1:483791
 ❯ rust_begin_unwind wasm:/wasm/00264a8e:1:474051
 ❯ core::panicking::panic_fmt::h14c85a61aa3d538e wasm:/wasm/00264a8e:1:477599
 ❯ core::option::expect_failed::h2966ad4c676148ac wasm:/wasm/00264a8e:1:474179
 ❯ verkle_trie::proof::VerkleProof::check::h8b40c80b7f75ebef wasm:/wasm/00264a8e:1:28591
 ❯ ffi_interface::verify_execution_witness::heca47ad411dad6cd wasm:/wasm/00264a8e:1:356435
 ```
 
 From Kev: it is failing due to an invariant failing 
 
 We can still make progress on the stateless client front with what we have here, but this will need fine-tuning in order to properly verify all blocks.  (I believe Kev is looking into this). The current version will at least give us some API to interact with and test.

**Pull request merged**